docs/4.0/i18n/zh-Hans/Development-Guidelines/resource-controller.md

```diff
@@ -0,0 +1,119 @@
+# Resource-controller如何编写
+
+resource-controller 是计量计费模块的一部分，请确保您提前了解[计量计费的设计](https://github.com/labring/sealos/blob/main/proposal/design/zh/metering%E8%AE%BE%E8%AE%A1.md)
+
+
+
+## resource-controller 是什么
+
+resource-controller 是计量计费模块的一部分，用来统计和计费资源的使用量，不同资源不同resource-controller，使用podresource-controller举例，podresource-controller会统计用户ns下pod的资源limit使用量，并且计算出价格，生成一个Resource CR.
+
+![](https://p.ipic.vip/p87a3h.png)
+
+
+
+## 怎么样编写resource-controller（以podresource-controller举例）
+
+### api定义
+
+```
+
+type PodResourceSpec struct {
+	ResourceName string                                          `json:"resourceName,omitempty"`
+	Resources    map[v1.ResourceName]meteringcommonv1.ResourcePrice `json:"resources,omitempty"`
+
+	// update used resources every Interval minutes
+	Interval int `json:"interval,omitempty"`
+}
+
+// PodResourceStatus defines the observed state of PodResource
+type PodResourceStatus struct {
+	LatestUpdateTime int64 `json:"latestUpdateTime,omitempty"`
+	SeqID            int64 `json:"seqID,omitempty"`
+}
+```
+
+seqID是每次podresource-controller统计计费完之后都需要+1，用当前seqID值来作为后缀命名创建的Resource CR，防止CR 名字一样。
+
+### 在reconcile 中实现以下接口
+
+https://github.com/labring/sealos/blob/main/pkg/metering/interface.go
```

Comment:
replace with relative file path

Reply:
(../../../../../pkg/metering/interface.go) is ok?